setup.py

```diff
@@ -0,0 +1,32 @@
+import os
+from setuptools import setup
+
+
+def create_tip_dir():
+    tip_dir = os.getenv('TIP_DIR') or os.path.join(os.path.expanduser('~'), 'tip')
+    os.makedirs(tip_dir, exist_ok=True)
+    return tip_dir
+
+
+def init_config(tip_dir):
+    from tip.config import pass_config
+
+    @pass_config
+    def _init_config(tip_dir, config):
+        config['tip_dir'] = tip_dir
```

Comment:
Why I think putting `tip_dir` in config is not a very good idea:
- It's configured twice, as env variable and in the config.
- Passing `tip_dir` everywhere makes the code less readable. Though, even having `tip_dir` in config it was not necessary to pass `tip_dir` everywhere.
- I'm not sure we know usage scenarios when user needs to configure it. But even if there are, one line like `tip_dir = os.getenv('TIP_DIR', '~/.tip')` solves the problem and simplifies everything.

Reply:
I don't think it is possible to set environment variable when VSCode runs `tipython`, however I'm currently researching it again. We used to have tip dir in environment variable before and I think we decided to use file because of VSCode integration.

Reply:
You can export env variable from .bashrc, .zshrc, etc like conda does. Though in our case we are good with default `os.getenv('TIP_DIR', '~/.tip')`, so you don't even need to export it. I guess you needed config to support environments that can't be switched through env variables.

Reply:
"when he can't install tip in default location" seems like practically unreal situation. Also you put the TIP config in home directory [here](https://github.com/itesoro/tip/pull/1/files#diff-93e09556c3223fedec13e1e3f850461411964c29872b362538af054d4d1db5a3R9), so the code already assumes that you can write there.
I don't mean that we shouldn't think about this use case. What I mean is it's just enough to make sure it can be covered in the future without serious refactoring.
My proposition is to put CLI scripts in `TIP_DIR/bin/`. AFAIK it can be done via setup.py. This way when you run a TIP command it will know the path to `TIP_DIR` without specifying it in config with hardcoded path or env variables.

Reply:
The practical use case is to make shared site-packages. This can be done just by specifying path `site-packages` in config.